projects/plugins/jetpack/extensions/blocks/sharing-buttons/view.js

```diff
@@ -23,6 +35,22 @@ if ( typeof window !== 'undefined' ) {
 				event.preventDefault();
 				event.stopPropagation();
 
+				if ( service === 'share' ) {
+					if ( link?.href && isWebShareAPIEnabled( { url: link.href } ) ) {
+						navigator.share( { url: link.href } );
+					} else {
+						const [ tooltip ] = document.getElementsByClassName( 'tooltiptext' );
+						if ( tooltip && tooltip.style ) {
+							tooltip.style.display = 'initial';
+							setTimeout( () => {
+								tooltip.style.display = 'none';
+							}, 2000 );
+						}
+						navigator?.clipboard?.writeText( link.href );
```

Comment:
I think we'd want to share the post URL without the extra query strings from the sharing button. Something like this?

```diff
diff --git a/projects/plugins/jetpack/extensions/blocks/sharing-buttons/view.js b/projects/plugins/jetpack/extensions/blocks/sharing-buttons/view.js
index 524eb9668c..fe52121ed2 100644
--- a/projects/plugins/jetpack/extensions/blocks/sharing-buttons/view.js
+++ b/projects/plugins/jetpack/extensions/blocks/sharing-buttons/view.js
@@ -36,8 +36,15 @@ if ( typeof window !== 'undefined' ) {
 				event.stopPropagation();
 
 				if ( service === 'share' ) {
-					if ( link?.href && isWebShareAPIEnabled( { url: link.href } ) ) {
-						navigator.share( { url: link.href } );
+					if ( ! link.href ) {
+						return;
+					}
+
+					const postUrl = new URL( link.href );
+					postUrl.search = '';
+					const postLink = postUrl.toString();
+					if ( isWebShareAPIEnabled( { url: postLink } ) ) {
+						navigator.share( { url: postLink } );
 					} else {
 						const [ tooltip ] = document.getElementsByClassName( 'tooltiptext' );
 						if ( tooltip && tooltip.style ) {
@@ -46,7 +53,7 @@ if ( typeof window !== 'undefined' ) {
 								tooltip.style.display = 'none';
 							}, 2000 );
 						}
-						navigator?.clipboard?.writeText( link.href );
+						navigator?.clipboard?.writeText( postLink );
 					}
 					return;
 				}
```

Reply:
I've updated the link but on PHP level, not to include params in URL. Do you think it would be ok?

Reply:
That may work indeed, I haven't tested yet.